packages/big-design/src/components/Select/Select.tsx

```diff
@@ -228,31 +190,13 @@ export class Select extends React.PureComponent<SelectProps, SelectState> {
   }
 
   private renderChips() {
-    const { children, multi, value } = this.props;
+    const { options, multi, value: values } = this.props;
 
-    if (!multi || !value) {
+    if (!multi || !values) {
       return;
     }
 
-    const listItems = React.Children.map(children, child => {
-      if (!React.isValidElement(child)) {
-        return;
-      }
-
-      return { value: child.props.value, text: child.props.children };
-    });
-
-    if (Array.isArray(value)) {
-      return value.map(val => {
-        const selectedItem = listItems.find(item => item && String(item.value) === String(val));
-
-        return selectedItem && selectedItem.text;
-      });
-    } else {
-      const listItem = listItems.find(item => item && String(item.value) === String(value));
-
-      return listItem && [listItem.text];
-    }
+    return Array.isArray(values) ? values.map(value => options.find(option => option.value === value)) : [];
```

Comment:
🍹You could default `values` to an empty `[]` and remove the `Array.isArray` check.

Reply:
Doesn't seem to work, since values can be of type `T` or `T[]`.